Bug: runserver --with-pages builds all targets including binaries for wasm32

## Prerequisites
- [x] I have searched existing issues to verify this bug hasn't been reported

## Bug Description
`runserver --with-pages` runs `cargo build --target wasm32-unknown-unknown` without `--lib` flag, causing it to attempt compiling binary targets (e.g., `manage.rs`) for the wasm32 target. Binary targets contain native-only imports (`reinhardt::commands`, `tokio`, etc.) that are correctly gated with `#[cfg(not(target_arch = "wasm32"))]` in reinhardt-web but are unconditionally imported in the binary.

### Expected behavior
`runserver --with-pages` should pass `--lib` flag when building WASM to only compile the library target.

### Actual behavior
The command runs `cargo build --target wasm32-unknown-unknown` without `--lib`, which tries to compile all targets including binaries that have native-only dependencies.

### Error output
```
error[E0432]: unresolved import `reinhardt::commands`
  --> dashboard/src/bin/manage.rs:11:16
```

### Environment
- reinhardt-web: 6437c1ce (main branch)
- Rust: stable
- OS: macOS

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/crates/reinhardt-commands/src/wasm_builder.rs
+++ b/crates/reinhardt-commands/src/wasm_builder.rs
@@ -247,6 +247,7 @@ impl WasmBuilder {
 	fn run_cargo_build(&self) -> Result<(), WasmBuildError> {
 		let mut cmd = Command::new("cargo");
 		cmd.arg("build")
+			.arg("--lib")
 			.arg("--target")
 			.arg("wasm32-unknown-unknown")
 			.current_dir(&self.config.project_dir);

